### PR TITLE
[2.0] Cleaned up sample app

### PR DESCRIPTION
**Pull request by gpeal**
Key changes:
* Use mvrx instead of mvrx-rxjava2. A new sample will be needed to demonstrate rxjava.
* Removed the feeling of coupling to Epoxy. Epoxy is still used for lists but other screens have been moved to ViewBinding.
* Added the Fragment and View ViewBinding delegates we've been using at Tonal. They have been working great for us.
* General code cleanup/simplification from learnings over the past couple of years.

**API ADDITION**
* I added Parcelable.asMvRxArgs() and Serializable.asMvRxArgs() that takes either class and returns a Bundle with the object set as `KEY_ARG`

@denis-bezrukov

### CODE REVIEW

File: mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxExtensions.kt [hunk 1]
```diff
@@ -154,6 +157,22 @@ fun <V : Any> args() = object : ReadOnlyProperty<Fragment, V> {
...
+/**
+ * Takes anything that is [Parcelable] and creates a MvRx Fragment argument [Bundle].
+ *
+ * Set this as your Fragment's arguments and you can use the [args] property delegate in your Fragment
+ * to easily retrieve it.
+ */
+fun Serializable.asMvRxArgs() = Bundle().apply { putSerializable(MvRx.KEY_ARG, this@asMvRxArgs) }
+
+/**
+ * Takes anything that is serializable and creates a MvRx Fragment argument [Bundle].
```
Comment (elihart) [inline mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxExtensions.kt:169]: you have the comments swapped ` that is serializable`

File: sample/src/main/java/com/airbnb/mvrx/sample/features/dadjoke/RandomDadJokeFragment.kt [hunk 2]
```diff
@@ -28,7 +35,9 @@ class RandomDadJokeViewModel(
     }
 
     fun fetchRandomJoke() {
-        dadJokeService.random().subscribeOn(Schedulers.io()).execute { copy(joke = it) }
+        suspend {
+            dadJokeService.random()
```
Comment (elihart) [inline sample/src/main/java/com/airbnb/mvrx/sample/features/dadjoke/RandomDadJokeFragment.kt:39]: nice :)

File: sample/src/main/java/com/airbnb/mvrx/sample/views/BasicRow.kt [hunk 3]
```diff
@@ -2,40 +2,35 @@ package com.airbnb.mvrx.sample.views
...
 @ModelView(autoLayout = ModelView.Size.MATCH_WIDTH_WRAP_HEIGHT)
 class BasicRow @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
 ) : LinearLayout(context, attrs, defStyleAttr) {
-
-    private val titleView: TextView
-    private val subtitleView: TextView
+    private val binding: BasicRowBinding by viewBinding()
```
Comment (elihart) [inline sample/src/main/java/com/airbnb/mvrx/sample/views/BasicRow.kt:19]: 👍

**Comment by gpeal** [comment]
@elihart Good idea. Would you be open to making a new example for that? I just kept hearing people worried about having to learn them together and I wanted to make it more clear that it wasn't necessary but having a helper would be nice too.

**Pull request merged**